ani-cli

```diff
@@ -1,131 +1,78 @@
+nth() {
+	stdin=$(cat -)
+	[ -z "$stdin" ] && return 1
+	line_count="$(printf "%s\n" "$stdin" | wc -l)"
+	[ "$line_count" -eq 1 ] && printf "%s" "$stdin" && return 0
+	line=$(printf "%s" "$stdin" | sed 's|.*\t||' | launcher "$*")
+	[ -n "$line" ] && printf "%s" "$stdin" | grep -w "${line}$" || exit 1
 }
 
-#display error message and exit
-die () {
-	err "$*"
+die() {
+	printf "%s\n" "$*" 1>&2
 	exit 1
 }
 
-# display an informational message (first argument in green, second in magenta)
-inf () {
-	printf "\33[2K\r\033[1;35m%s \033[1;35m%s\033[0m\n" "$1" "$2"
+help_info() {
+	printf "
+    Usage: 
+    %s [options] [query]
+    %s [query] [options]
+    %s [options] [query] [options]
+
+    Options:
+      -c, --continue
+        Continue watching from history
+      -d, --download
+        Download the video instead of playing it
+      -s, --syncplay
+        Use Syncplay to watch with friends
+      -q, --quality
+        Specify the video quality
+      --vlc
+        Use VLC to play the video
+      -v, -V, --version
+        Show the version of the script
+      -h, --help
+        Show this help message and exit
+      -e, --episode
+        Specify the number of episodes to watch
+      -u, -U, --update
+        Update the script
+    Some example usages:
+      %s -q 720p banana fish
+      %s -d -e 2 cyberpunk edgerunners
+      %s --vlc cyberpunk edgerunners -q 1080p -e 4
+    \n" "${0##*/}" "${0##*/}" "${0##*/}" "${0##*/}" "${0##*/}" "${0##*/}"
 }
 
-progress() {
-	printf "\33[2K\r\033[1;34m%s\033[0m" "$1"
-}
-
-debug () {
-	printf "\n\033[1;32mReferrer :\033[0m %s\n\033[1;32mlinks >>\n\033[0m%s\n" "$1" "$2"
-}
-
-# prompts the user with message in $1-2 ($1 in blue, $2 in magenta) and saves the input to the variables in $REPLY and $REPLY2
-prompt () {
-	[ -n "$*" ] && printf "\33[2K\r\033[1;35m%s\n" "$*"
-	printf "\033[1;35m> \033[0m"
-	read -r REPLY REPLY2
-}
-
-selection_menu() {
-	menu_line_parity=0
-	while read -r option line; do
-		if [ "$option" = "q" ]; then
-			printf "\033[1;31m(\033[1;31m%s\033[1;31m) \033[1;31m%s\033[0m\n" "$option" "$line"
-		else
-			if [ "$menu_line_parity" -eq 0 ]; then
-				printf "\033[1;33m(\033[1;33m%s\033[1;33m) \033[1;33m%s\033[0m\n" "$option" "$line"
-				menu_line_parity=1
-			else
-				printf "\033[1;36m(\033[1;36m%s\033[1;36m) \033[1;36m%s\033[0m\n" "$option" "$line"
-				menu_line_parity=0
-			fi
-		fi
-	done <<-EOF
-	$*
-	EOF
-	prompt
-}
-
-selection_menu_fzf() {
-	printf "%s\n%s" "$1" "$2" | fzf --height=30% --border -1 --layout=reverse --header-first --header-lines=1 --cycle --with-nth 2.. | cut -d\  -f1
-}
-
-version_text () {
-	inf "Version: $VERSION"
+version_info() {
+	printf "%s\n" "$version_number"
+	exit 0
 }
 
-# get the newest version of this script from github and replace it
-update_script () {
+update_script() {
 	update="$(curl -A "$agent" -s "https://raw.githubusercontent.com/pystardust/ani-cli/master/ani-cli")" || die "Connection error"
```

Comment:
Mirgate to `wget`, since this is the last occurence of `curl`

Reply:
okay